### DISCUSSION TITLE
Ability to do a radius search to find POIS

### DISCUSSION THREAD

**Discussion by Sunraj751** [valhalla/valhalla#4654] [Q&A] [answered]
Hi, I was wondering if Valhalla's locate service is able to do a radius search where it finds a specific POI within a certain radius?

Example: Given an (X,Y) coordinate, find a restaurant within 10Kms.

**Comment by nilsnolde** [comment, marked as answer]
With our landmarks we _could_ have functionality like that, but we don’t, and likely won’t in the future. That’s what a geocoder should provide, check out pelias.

**Reply by kevinkreiser** [reply]
imho locate will (but doesn't yet) return landmarks on the edges it returns but @nilsnolde is still right. use a geocoder

**Reply by nilsnolde** [reply]
true. you're right it'll be part of /locate output, just not in the way the OP would like it (with type filtering).

**Reply by Sunraj751** [reply]
Thanks for the response @nilsnolde, @kevinkreiser. Yeah I did look into pelias, but was having a hard time trying to figure out hosting aspect of it. Will look into it again. 

If you guys have any tips or tricks around that could help with hosting pelias instance as a containerized application that would be greatly appreciated !!

**Reply by nilsnolde** [reply]
it's a complex data ingestion pipeline in pelias, but IMO they make everything insanely easy: https://github.com/pelias/docker. certainly much much better documented than valhalla :sweat_smile:

**Reply by Sunraj751** [reply]
No doubt it's very well documented but yeah the data ingestion pipelines makes it harder, cause when I run docker, it's like 10 containers spinning up, which is like confusing (at least for a docker beginner like me). I think for me valhalla's docker was easier to setup, as all i had to do was set the osm link and done, super easy and neat!
